src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.Fixer.cs

```diff
@@ -26,11 +28,123 @@ public sealed override FixAllProvider GetFixAllProvider()
+                        (documentEditor, syntaxNode, ct) => Task.Run(() => documentEditor.SetAccessibility(badVisibilityNode, Accessibility.Public))).ConfigureAwait(false);
+                    var newDocument = symbolEditor.GetChangedDocuments().Single();
+                    var newRoot = await newDocument.GetSyntaxRootAsync(context.CancellationToken);
+                    return context.Document.WithSyntaxRoot(newRoot);
+                default:
+                    return context.Document;
+            }
+        }
+
+        private static ExpectedMethodSignature GetExpectedMethodSignature(IMethodSymbol operatorOverloadSymbol, Compilation compilation)
+        {
+            var containingType = (ITypeSymbol)operatorOverloadSymbol.ContainingType;
+            var returnType = operatorOverloadSymbol.ReturnType;
+            var expectedName = OperatorOverloadsHaveNamedAlternatesAnalyzer.GetExpectedAlternateMethodGroup(operatorOverloadSymbol.Name, returnType).AlternateMethod1;
+            switch (operatorOverloadSymbol.Name)
+            {
+                case "op_GreaterThan":
+                case "op_GreaterThanOrEqual":
+                case "op_LessThan":
+                case "op_LessThanOrEqual":
+                    // e.g., public int CompareTo(MyClass other)
+                    var intType = compilation.GetSpecialType(SpecialType.System_Int32);
+                    return new ExpectedMethodSignature(expectedName, intType, ImmutableArray.Create(Tuple.Create("other", containingType)), isStatic: false);
+                case "op_Decrement":
+                case "op_Increment":
+                case "op_UnaryNegation":
+                case "op_UnaryPlus":
+                    // e.g., public MyClass Decrement(MyClass item)
+                    return new ExpectedMethodSignature(expectedName, returnType, ImmutableArray.Create(Tuple.Create("item", containingType)));
+                case "op_Implicit":
+                    // e.g., public int ToInt32()
+                    return new ExpectedMethodSignature(expectedName, returnType, ImmutableArray.Create<Tuple<string, ITypeSymbol>>(), isStatic: false);
+                default:
+                    // e.g., public static MyClass Add(MyClass left, MyClass right)
+                    return new ExpectedMethodSignature(expectedName, returnType, ImmutableArray.Create(Tuple.Create("left", containingType), Tuple.Create("right", containingType)));
+            }
+        }
+
+        private class ExpectedMethodSignature
+        {
+            public string Name { get; }
+            public ITypeSymbol ReturnType { get; }
+            public IEnumerable<Tuple<string, ITypeSymbol>> Parameters { get; }
+            public bool IsStatic { get; }
+
+            public ExpectedMethodSignature(string name, ITypeSymbol returnType, IEnumerable<Tuple<string, ITypeSymbol>> parameters, bool isStatic = true)
```

Comment:
It's not obvious to me that there is a reasonable default for `isStatic`. I'd remove the default and specify it explicitly everywhere.